web/src/user_topics.js

```diff
@@ -11,54 +11,67 @@ import * as stream_data from "./stream_data";
 import * as timerender from "./timerender";
 import {get_time_from_date_muted} from "./util";
 
-const muted_topics = new Map();
+const topics_dict = new Map();
 
-export const visibility_policy = {
-    INHERIT: 0,
-    MUTED: 1,
-    UNMUTED: 2,
-    FOLLOWED: 3,
-};
-
-export function add_muted_topic(stream_id, topic, date_muted) {
-    let sub_dict = muted_topics.get(stream_id);
+export function update_topics_dict(stream_id, topic, visibility_policy, date_updated) {
+    let sub_dict = topics_dict.get(stream_id);
     if (!sub_dict) {
```

Comment:
This will have incorrect behavior when `!sub_dict is true` and `visibility_policy=INHERIT`

Reply:
Refactored this function to have the correct behavior as before in the updated PR.